server/src/main/java/org/elasticsearch/search/lookup/SourceLookup.java

```diff
@@ -52,6 +53,7 @@
     private Map<String, Object> source;
     private XContentType sourceContentType;
 
+    @Override
     public Map<String, Object> source() {
         return source;
```

Comment:
I wonder if this should now `loadSourceIfNeeded`?

Reply:
++, it should.  Previously we would check against `source()` to see if it was null or not, but now we can check against `Source.EMPTY_SOURCE` directly.

I think this means we can also make `loadSourceIfNeeded` private and not have to call it from scripts, but that's a bigger change so I'd like to leave it for a followup.